service/api/service_impl.go

```diff
@@ -149,6 +151,117 @@ func (s serviceImpl) PublishToLocalQueue(
 	return nil
 }
 
+func (s serviceImpl) Rpc(
+	ctx context.Context, request xdbapi.ProcessExecutionRpcRequest,
+) (response *xdbapi.ProcessExecutionRpcResponse, retErr *ErrorWithStatus) {
+	latestPrcExe, err := s.store.GetLatestProcessExecution(ctx, data_models.GetLatestProcessExecutionRequest{
+		Namespace: request.GetNamespace(),
+		ProcessId: request.GetProcessId(),
+	})
+	if err != nil {
+		return nil, s.handleUnknownError(err)
+	}
+
+	if latestPrcExe.NotExists {
+		return nil, NewErrorWithStatus(http.StatusNotFound, "Process does not exist")
+	}
+
+	iwfWorkerBaseUrl := urlautofix.FixWorkerUrl(latestPrcExe.WorkerUrl)
+	apiClient := xdbapi.NewAPIClient(&xdbapi.Configuration{
+		Servers: []xdbapi.ServerConfiguration{
+			{
+				URL: iwfWorkerBaseUrl,
+			},
+		},
+	})
+
+	loadGlobalAttributeResponse := xdbapi.LoadGlobalAttributeResponse{}
+	if latestPrcExe.GlobalAttributeConfig != nil {
+		loadGlobalAttrResp, err := s.store.LoadGlobalAttributes(ctx, data_models.LoadGlobalAttributesRequest{
+			TableConfig: *latestPrcExe.GlobalAttributeConfig,
+			Request:     request.GetLoadGlobalAttributesRequest(),
+		})
+		if err != nil {
+			return nil, s.handleUnknownError(err)
+		}
+
+		loadGlobalAttributeResponse = loadGlobalAttrResp.Response
+	}
+
+	workerApiCtx, cancF := s.createContextWithTimeoutForRpc(ctx, request.GetTimeoutSeconds())
+	defer cancF()
+
+	req := apiClient.DefaultAPI.ApiV1XdbWorkerProcessRpcPost(workerApiCtx)
+	resp, httpResp, err := req.ProcessRpcWorkerRequest(
+		xdbapi.ProcessRpcWorkerRequest{
+			Context: xdbapi.Context{
+				ProcessId:               request.GetProcessId(),
+				ProcessExecutionId:      latestPrcExe.ProcessExecutionId.String(),
+				ProcessStartedTimestamp: latestPrcExe.StartTimestamp,
+			},
+			ProcessType:            latestPrcExe.ProcessType,
+			RpcName:                request.GetRpcName(),
+			Input:                  request.Input,
+			LoadedGlobalAttributes: &loadGlobalAttributeResponse,
+		},
+	).Execute()
+	if httpResp != nil {
+		defer httpResp.Body.Close()
+	}
+
+	if utils.CheckHttpResponseAndError(err, httpResp, s.logger) {
+		return nil, NewErrorWithStatus(
```

Comment:
There could be more useful info that the caller need to get from the worker execution. 

In the API, we have decided a workerErrorResponse, and we should try to extract it from the http body to caller. 

See similar in iWF server: https://github.com/indeedeng/iwf/blob/main/service/common/rpc/invoke.go#L59

The reason is that worker may want to return some useful error type for caller to make different decision on.

Reply:
Let's make another PR to handle it. Several places need to modify accordingly, not only here. Created https://github.com/xcherryio/xcherry/issues/98.